### PR TITLE
Manage upload to ipfs

### PR DESCRIPTION
**Pull request by Ugo**
The idea here is to publish the results depending on the beneficiary of the task:
- if the beneficiary is null (meaning the result is public), the result will be pushed to the ipfs node defined in the core.
- if the beneficiary is not null, the result will be pushed in the mongo repo of the result repo (same as before)

Some refactoring will be needed to split the replicate update status into several entrypoints instead of a single one at the moment.

### CODE REVIEW

File: src/main/java/com/iexec/core/replicate/ReplicatesService.java [hunk 1]
```diff
@@ -346,10 +360,24 @@ private boolean isTaskStatusFailOnChain(String chainTaskId, String walletAddress
         }
     }
 
+    boolean isPublicResult(String chainTaskId, Integer chainId) {
```
Comment (jeremyjams) [inline src/main/java/com/iexec/core/replicate/ReplicatesService.java]: Can we move `isPublicResult` to common? (We're gonna need it for the seperate Result Repo)

File: src/main/java/com/iexec/core/result/ResultService.java [hunk 2]
```diff
@@ -103,22 +139,16 @@ boolean isOwnerOfResult(Integer chainId, String chainTaskId, String downloaderAd
         return true;
     }
 
+    boolean isPublicResult(String chainTaskId) {
+        return isPublicResult(chainTaskId, 0);
+    }
+
     boolean isPublicResult(String chainTaskId, Integer chainId) {
```
Comment (jeremyjams) [inline src/main/java/com/iexec/core/result/ResultService.java]: We have a duplicate here

File: src/main/java/com/iexec/core/result/ipfs/IPFSService.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        Multihash filePointer = Multihash.fromBase58(ipfsHash);
+        try {
+            return Optional.of(ipfs.get(filePointer));
+        } catch (IOException e) {
+            log.error("Error when trying to retrieve ipfs object [hash:{}]", ipfsHash);
+        }
+        return Optional.empty();
+    }
+
+    public boolean doesContentExists(String ipfsHash) {
```
Comment (jeremyjams) [inline src/main/java/com/iexec/core/result/ipfs/IPFSService.java]: doesContentExist* (no 's')!

File: src/main/java/com/iexec/core/replicate/ReplicatesService.java [hunk 4]
```diff
@@ -170,6 +169,16 @@ private int getNbReplicatesWithGivenStatusJustBeforeWorkerLost(String chainTaskI
         return Optional.empty();
     }
 
+    public Optional<Replicate> getReplicateWithResultUploadedStatus(String chainTaskId) {
```
Comment (jeremyjams) [inline src/main/java/com/iexec/core/replicate/ReplicatesService.java]: It seems we have an issue when worker is lost after UPLOADED

Comment (jeremyjams) [reply src/main/java/com/iexec/core/replicate/ReplicatesService.java]: https://github.com/iExecBlockchainComputing/iexec-core/blob/301656cd5c2353d5c131c95c330717bba388715a/src/main/java/com/iexec/core/replicate/ReplicatesService.java#L148

File: src/main/java/com/iexec/core/replicate/ReplicatesService.java (hunk 1 above)
Comment (Ugo) [reply src/main/java/com/iexec/core/replicate/ReplicatesService.java]: right, moved to common.

File: src/main/java/com/iexec/core/result/ResultService.java (hunk 2 above)
Comment (Ugo) [reply src/main/java/com/iexec/core/result/ResultService.java]: removed.

File: src/main/java/com/iexec/core/result/ipfs/IPFSService.java (hunk 3 above)
Comment (Ugo) [reply src/main/java/com/iexec/core/result/ipfs/IPFSService.java]: thanks, changed.

File: src/main/java/com/iexec/core/replicate/ReplicatesService.java (hunk 4 above)
Comment (Ugo) [reply src/main/java/com/iexec/core/replicate/ReplicatesService.java]: great catch, thanks! I've modified the code.

**Pull request merged**